src/frontend/Typechecker.ml

```diff
@@ -1294,7 +1302,9 @@ let returntype_leastupperbound loc rt1 rt2 =
    |ReturnType UInt, ReturnType UReal ->
       UnsizedType.ReturnType UReal
   | _, _ when rt1 = rt2 -> rt2
-  | _ -> Semantic_error.mismatched_return_types loc rt1 rt2 |> error
+  | _ ->
+      (* should no longer be possible with improved return statement type checking *)
+      Semantic_error.mismatched_return_types loc rt1 rt2 |> error
```

Comment:
Now that these checks are redundant, `statement_returntype` does not need to keep track of the type and can be simplified: `AnyReturnType` and `Complete` are equivalent, as are `NoReturnType` and `Incomplete`.

And for #1340 you can define
```ocaml
type statement_returntype =
  | Incomplete of bool
  | Complete
```
where the boolean indicates whether there are any `break`s present.

Reply:
I went with a 3-part enum rather than adding the bool, 
```ocaml
type statement_returntype =
  | Incomplete
  | UnclearControlFlow (* is any break present *)
  | Complete
```
but otherwise agree this is much nicer!

Reply:
I wouldn't call it `UnclearControlFlow`.
`Complete` means any possible execution of the statement must exit the function, either through a `return` or `reject`. `Incomplete` means some possible execution passes control flow to the following statement. The third option is more like `NonlocalControlFlow` meaning the control flow can jump out of this statement into some later statement in the same function (i.e. `break` jumps to the end of the loop, wherever that may be. But we don't consider a `continue` to "nonlocal" for the purposes of `while` completeness checking.)

Reply:
I changed the name and wrote a longer docstring based on your comment